decode/evm_rpc.go

```diff
@@ -248,13 +249,13 @@ func lookupBlockNumberFromHashParam(ctx context.Context, evmClient *ethclient.Cl
 		return 0, fmt.Errorf(fmt.Sprintf("error decoding block hash param from params %+v at index %d", params, paramIndex))
 	}
 
-	block, err := evmClient.BlockByHash(ctx, common.HexToHash(blockHash))
+	header, err := evmClient.HeaderByHash(ctx, common.HexToHash(blockHash))
```

Comment:
as mentioned in slack, the call to `BlockByHash` was failing when querying a hash that did not exist in the chain and causing the service to panic. i confirmed the problem on this branch, master, and on the commit of the currently deployed public proxy (547e9ef).

it doesn't appear to be a real problem in production, as the service properly responds when you query a non-existent block hash.

however, i propose changing the `BlockByHash` call to `HeaderByHash`. `BlockByHash` actually makes multiple requests to get the header and the txs. we only need the block number, so header is sufficient.

CC @evgeniy-scherbina this will conflict with your evm client changes in #39